engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -842,6 +878,707 @@ static int SetTexture(lua_State* L)
+ *
+ * * `fps`
+ * : [type:integer] optional fps of the animation, the default value is 30
+ *
+ * * `flip_vertical`
+ * : [type:boolean] optional flip the animation vertically, the default value is false
+ *
+ * * `flip_horizontal`
+ * : [type:boolean] optional flip the animation horizontally, the default value is false
+ *
+ * * `geometries`
+ * : [type:table] A list of the geometries that should map to the texture data. Supports the following fields:
+ *
+ * * `vertices`
+ * : [type:table] a list of the vertices in texture space of the geometry in the form {px0, py0, px1, py1, ..., pxn, pyn}
+ *
+ * * `uvs`
+ * : [type:table] a list of the uv coordinates in texture space of the geometry in the form of {u0, v0, u1, v1, ..., un, vn}
+ *
+ * * `indices`
+ * : [type:table] a list of the indices of the geometry in the form {i0, i1, i2, ..., in}. Each tripe in the list represents a triangle.
+ *
+ * @note The index values are zero based where zero refers to the first entry of the vertex and uv lists
+ *
+ * @examples
+ * Add a new animation to an existing atlas
+ *
+ * ```lua
+ * function init(self)
+ *     local data = resource.get_atlas("/main/my_atlas.a.texturesetc")
+ *     local my_animation = {
+ *         id          = "my_new_animation",
+ *         width       = 128,
+ *         height      = 128,
+ *         frame_start = 1,
+ *         frame_end   = 6,
+ *         playback    = go.PLAYBACK_LOOP_PINGPONG,
+ *         fps         = 8
+ *     }
+ *     table.insert(data.animations, my_animation)
+ *     resource.set_atlas("/main/my_atlas.a.texturesetc", data)
+ * end
+ * ```
+ *
+ * @examples
+ * Sets atlas data for a 256x256 texture with a single animation being rendered as a quad
+ *
+ * ```lua
+ * function init(self)
+ *     local params = {
+ *         texture = "/main/my_256x256_texture.texturec",
+ *         animations = {
+ *             {
+ *                 id          = "my_animation",
+ *                 width       = 256,
+ *                 height      = 256,
+ *                 frame_start = 1,
+ *                 frame_end   = 2,
+ *             }
+ *         },
+ *         geometries = {
+ *             {
+ *                 vertices = {
+ *                     0,   0,
+ *                     0,   256,
+ *                     256, 256,
+ *                     256, 0
+ *                 },
+ *                 uvs = {
+ *                     0, 0,
+ *                     0, 256,
+ *                     256, 256,
+ *                     256, 0
+ *                 },
+ *                 indices = { 0,1,2,0,2,3 }
+ *             }
+ *         }
+ *     }
+ *     resource.set_atlas("/main/test.a.texturesetc", params)
+ * end
+ * ```
+ */
+
+static int SetAtlas(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 0);
+
+    dmhash_t path_hash = dmScript::CheckHashOrString(L, 1);
+    void* atlas_res    = CheckResource(L, g_ResourceModule.m_Factory, path_hash, "texturesetc");
+    assert(atlas_res);
```

Comment:
Do we need this, if we just did a CheckResource?

Reply:
No, not really. Removing it